src/openforms/appointments/base.py

```diff
@@ -185,13 +190,26 @@ def get_required_customer_fields(
         """
         raise NotImplementedError()
 
+    def normalize_contact_details(
+        self, data: dict[F, JSONPrimitive]
+    ) -> dict[F, JSONPrimitive]:
+        # XXX: We could consider using openforms.formio.service.normalize_value_for_component,
+        # but there currently is no phone normalizer and adding it could potentially break this
+        # component...
+        normalized = {}
+        for key, value in data.items():
+            for normalizer in self.normalizers.get(key, []):
+                value = normalizer(value)
+            normalized[key] = value
+        return normalized
```

Comment:
For this beauty, Haskell has the `.` operator and PureScript `<<<`. You could read it as "after".
So `f . g` is another function that applies `f` to the result of applying `g` to its argument.  Instead of a list of functions to loop through, the normalizer would be 1 function to apply... And the default not `[]` but the identity function. And this checks, because we defined `Normalizer` as a function from `a` to `a`; a functor... more specifically an endofunctor... You know... the kind of functors from the definition of an monad. :wink: